microcosm_flask/conventions/relation.py

```diff
@@ -77,9 +76,9 @@ def register_search_relation_endpoint(graph, obj, path_prefix, func, request_sch
     :param response_schema: a marshmallow schema to encode (a single) response item
     """
 
-    paginated_list_schema = make_paginated_list_schema(obj[1], response_schema)()
+    paginated_list_schema = make_paginated_list_schema(ns.object_ns, response_schema)()
```

Comment:
Note the `ns.object_ns` argument. Here's what's happening:
 - The pagination and linking code now depend on `Namespaces`, so they expect us to pass one
 - For relation-oriented endpoints, we have a subject, verb, object construction (e.g. `folder.search_for_files`)
 - The (paginated) result type of `search_for` operation is the object, not the subject (e.g. returns a page of `files`, not `folders`)
 - The `Namespace` knows how to perform this transformation